common/network-common/pom.xml

```diff
@@ -66,7 +66,7 @@
     </dependency>
     <dependency>
       <groupId>org.apache.spark</groupId>
-      <artifactId>spark-test-tags_${scala.binary.version}</artifactId>
+      <artifactId>spark-tags_${scala.binary.version}</artifactId>
```

Comment:
Basically, what you have to do is: everywhere you're changing this, you need to also add `<classifier>tests</classifier>`. In modules where both the test and non-test tags are used, you need both dependencies (with and without classifier).

It might be easier to just leave the test tags in the main source base...